### PR TITLE
H-2823: Add serde implementation for non-string hash-map keys

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Non-string keys are not allowed in JSON. This adds a serialization to convert keys to string for a `HashMap`.

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

**Review by indietyp — APPROVED** [review]
Great stuff!

File: libs/@local/codec/src/serde/string_hash_map.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+/// The default serializer for [`HashMap`].
+///
+/// This is only here to allow the use of `#[serde(with = "string_hash_map")]`.
+///
+/// # Errors
+///
+/// This function can return any error that can be returned by serializing a [`HashMap`].
+#[inline]
+pub fn serialize<K, V, H, S>(map: &HashMap<K, V, H>, serializer: S) -> Result<S::Ok, S::Error>
```
Comment (indietyp) [inline libs/@local/codec/src/serde/string_hash_map.rs:57]: Wouldn't that lead to an asymmetrical encoding? What if the key doesn't coerce to a string? Should display or smth be used here?

**Review by indietyp — CHANGES_REQUESTED** [review]
Sorry some overlooked changes looked at the deserialization code first asking for clarification

File: libs/@local/codec/src/serde/string_hash_map.rs (hunk 1 above)
Comment (TimDiekmann) [reply libs/@local/codec/src/serde/string_hash_map.rs:57]: I'm not sure if we want to add the overhead of converting it to a string. The implementation of `HashMap` is already asymmetrical. It's serializing to a string as key, but does not deserialize from a string to e.g. `usize`.

Comment (indietyp) [reply libs/@local/codec/src/serde/string_hash_map.rs:57]: I see yea then that's fine. Makes sense

**Review by indietyp — APPROVED** [review]

**Review by indietyp — DISMISSED** [review]

**Review by indietyp — APPROVED** [review]

**Pull request merged**